File: csharp/Solitaire/DeckTests/SuitTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Silnith.Game.Deck.Tests
{
    [TestClass]
    public class SuitTests
    {
        [TestMethod]
        public void TestGetColorClub()
        {
            Assert.AreEqual(Color.Black, Suit.Club.GetColor());
        }

        [TestMethod]
        public void TestGetColorDiamond()
        {
            Assert.AreEqual(Color.Red, Suit.Diamond.GetColor());
        }

        [TestMethod]
        public void TestGetColorHeart()
        {
            Assert.AreEqual(Color.Red, Suit.Heart.GetColor());
        }

        [TestMethod]
        public void TestGetColorSpade()
        {
            Assert.AreEqual(Color.Black, Suit.Spade.GetColor());
        }

        [TestMethod]
        public void TestToStringClub()
        {
            Assert.AreEqual("Club", Suit.Club.ToString());
        }

        [TestMethod]
        public void TestToStringDiamond()
        {
            Assert.AreEqual("Diamond", Suit.Diamond.ToString());
        }

        [TestMethod]
        public void TestToStringHeart()
        {
            Assert.AreEqual("Heart", Suit.Heart.ToString());
        }

        [TestMethod]
        public void TestToStringSpade()
        {
            Assert.AreEqual("Spade", Suit.Spade.ToString());
        }
    }
}

File: cpp/Solitaire/DeckTests/suit_tests.cpp
#include "CppUnitTest.h"

#include <silnith/game/deck/suit.h>

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

using namespace silnith::game::deck;

using namespace std::literals::string_literals;

template<>
std::wstring Microsoft::VisualStudio::CppUnitTestFramework::ToString<>(color const& color)
{
	return to_wstring(color);
}

template<>
std::wstring Microsoft::VisualStudio::CppUnitTestFramework::ToString<>(suit const& suit)
{
	return to_wstring(suit);
}

namespace DeckTests
{
	TEST_CLASS(SuitTests)
	{
	public:

		TEST_METHOD(TestGetColorClub)
		{
			Assert::AreEqual(color::black, get_color(suit::club));
		}

		TEST_METHOD(TestGetColorDiamond)
		{
			Assert::AreEqual(color::red, get_color(suit::diamond));
		}

		TEST_METHOD(TestGetColorHeart)
		{
			Assert::AreEqual(color::red, get_color(suit::heart));
		}

		TEST_METHOD(TestGetColorSpade)
		{
			Assert::AreEqual(color::black, get_color(suit::spade));
		}

		TEST_METHOD(TestToStringClub)
		{
			Assert::AreEqual("club"s, to_string(suit::club));
		}

		TEST_METHOD(TestToStringDiamond)
		{
			Assert::AreEqual("diamond"s, to_string(suit::diamond));
		}

		TEST_METHOD(TestToStringHeart)
		{
			Assert::AreEqual("heart"s, to_string(suit::heart));
		}

		TEST_METHOD(TestToStringSpade)
		{
			Assert::AreEqual("spade"s, to_string(suit::spade));
		}

		TEST_METHOD(TestToStringInvalidValue)
		{
			Assert::AreEqual("17"s, to_string(static_cast<suit>(17)));
		}

		TEST_METHOD(TestToWStringClub)
		{
			Assert::AreEqual(L"\u2663"s, to_wstring(suit::club));
		}

		TEST_METHOD(TestToWStringDiamond)
		{
			Assert::AreEqual(L"\u2666"s, to_wstring(suit::diamond));
		}

		TEST_METHOD(TestToWStringHeart)
		{
			Assert::AreEqual(L"\u2665"s, to_wstring(suit::heart));
		}

		TEST_METHOD(TestToWStringSpade)
		{
			Assert::AreEqual(L"\u2660"s, to_wstring(suit::spade));
		}

		TEST_METHOD(TestToWStringInvalidValue)
		{
			Assert::AreEqual(L"17"s, to_wstring(static_cast<suit>(17)));
		}
	};
}
